packages/ketcher-react/src/script/editor/tool/enhanced-stereo.js

```diff
@@ -1,76 +1,48 @@
-import { Pile } from 'ketcher-core'
-
 import { fromAtomsAttrs } from '../actions/atom'
 import { fromStereoFlagUpdate } from '../actions/fragment'
 import { findStereoAtoms } from '../actions/utils'
 
 function EnhancedStereoTool(editor) {
   if (!(this instanceof EnhancedStereoTool)) {
     const selection = editor.selection()
-    if (!selection || !selection.atoms) return null
 
-    const stereoAtoms = findStereoAtoms(editor.struct(), selection.atoms) // Map <aid,stereoLabel>
+    const stereoAtoms = findStereoAtoms(
+      editor.struct(),
+      selection
+        ? selection.atoms || []
+        : Array.from(editor.struct().atoms.keys())
+    )
+
     if (stereoAtoms.length === 0) return null
 
-    if (checkSelectionForFragment(editor, selection)) {
-      changeFragmentStereoAction(editor, stereoAtoms).then(
-        action => action && editor.update(action)
-      )
-    } else {
-      changeAtomsStereoAction(editor, stereoAtoms).then(
-        action => action && editor.update(action)
-      )
-    }
+    changeAtomsStereoAction(editor, stereoAtoms).then(
+      action => action && editor.update(action)
+    )
   }
 }
 
 function changeAtomsStereoAction(editor, stereoAtoms) {
   const struct = editor.struct()
   const restruct = editor.render.ctab
+  const stereoLabels = stereoAtoms.map(x => struct.atoms.get(x).stereoLabel)
```

Comment:
Please rename x to more informative param. NRE potentially can be thrown here so please add check for null